Section focus handling (#286)
wpaccessibility/wp-a11y-docs: docs/topics/code/focus-handling/input-hover.md

---

> title: On input and on hover

Reviewer: Shouldn't this be "On focus and on hover"?

Reviewer: I can see that you're handling both 'on input' and 'on focus and hover' in the same article. They are very similar, so that kind of makes sense. But probably need to mention all three in the title, then.

A **context** change unexpectedly changes the layout, information, keyboard focus, or functionality. For example, by immediately submitting a form after choosing a select option, after which the user is redirected to another page.

**Context** change is not the same as **content** change. Instantly adjusting search results, while keeping the user on the same page, is perfectly fine because it's expected.

{: .info .callout}
**Note**: When a dialog opens unexpectedly without user interaction, for example, with an advertisement or a request to sign up for a newsletter, is not a WCAG violation. But it's still confusing for screen reader users and annoying for everyone.

## How to test for behavior on input and on hover

First, determine if there are any interactive elements on the page, such as form fields, buttons, or links. Then:

- Tab through each interactive element with the keyboard, also give keyboard focus to any radio buttons and select options.
- Click on interactive elements with the mouse, including any select options.

> - Test for unexpected behavior, such as submitting the form, shifting focus, or opening a new window or tab.

Reviewer: ```suggestion
- Watch for unexpected behavior, such as submitting the form, shifting focus, or opening a new window or tab.
```

Is there a form on the page?

- Select/activate each form element.
- If applicable: fill out a form field and then leave it (remove focus).
- Test whether any unexpected behavior occurs, such as submitting the form without using the submit button, shifting focus, or opening a new window or tab.
- Test this with the mouse and then with the keyboard.

## Examples

{: .callout .dont }
**Don't**: Surprise your user by opening a dialog unexpectedly after hovering over or tabbing to a button.

```html
<button 
        onfocus="toggleDialog('dialog')" 
        onmousemove="toggleDialog('dialog')">
    On Sale!
</button>
```

{: .callout .do }
**Do**: Give the user the choice if they want to open a dialog.

```html
<button onclick="toggleDialog('dialog')">On Sale!</button>
```

{: .callout .dont }

> **Don't**: Already submit the form after the user selects an option.

Reviewer: ```suggestion
**Don't**: Submit the form as soon as the user selects an option.
```
